tests/integration/test_pipeline.py

```diff
@@ -320,3 +356,125 @@ def test_concatenations_when_csv_contains_endpoints_and_resources(tmp_path):
 
     assert "i-field3" in concats["o-field2"]["fields"]
     assert "i-field4" in concats["o-field2"]["fields"]
+
+
+def test_lookups_load_csv_success(
+    pipeline_dir,
+):
+    """
+    test csv loading functionality
+    :param pipeline_dir:
+    :return:
+    """
+    lookups = Lookups(pipeline_dir)
+    lookups.load_csv()
+
+    expected_num_entries = 1
+    assert len(lookups.entries) == expected_num_entries
+
+
+def test_lookups_load_csv_failure(
+    empty_pipeline_dir,
+):
+    """
+    test csv loading functionality when file absent
+    :param pipeline_dir:
+    :return:
+    """
+    lookups = Lookups(empty_pipeline_dir)
+
+    with pytest.raises(FileNotFoundError) as fnf_err:
+        lookups.load_csv()
+
+    expected_exception = "No such file or directory"
+    assert expected_exception in str(fnf_err.value)
+
+
+def test_lookups_get_max_entity_success(
+    pipeline_dir,
+):
+    """
+    test entity num generation functionality
+    :param pipeline_dir:
+    :return:
+    """
+    pipeline_name = "ancient-woodland"
+
+    lookups = Lookups(pipeline_dir)
+    lookups.load_csv()
+
+    expected_next_entity_num = 1
+    assert lookups.entity_num_gen.next() is expected_next_entity_num
+
+    max_entity_num = lookups.get_max_entity(pipeline_name)
+    lookups.entity_num_gen.state["current"] = max_entity_num
+    lookups.entity_num_gen.state["range_max"] = max_entity_num + 10
+
+    expected_next_entity_num = 1234568
+    assert lookups.entity_num_gen.next() == expected_next_entity_num
+
+
```

Comment:
adding an entry doesn't require a csv and hence can be a unit test. this feels like duplication

Reply:
It is very similar to the unit test.
The idea was that this one tests the functionality using file I/O.

Reply:
see above comment

Reply:
I agree they seem very similar.
I put both tests in - one for in-memory, one to ensure it's getting the last entity number from teh correct place in a file.
That is the only difference, though.

Reply:
I think remove them. the load_csv tests will test reading in the data. The unit test will test getting the max entity out.

Reply:
done